std/string.d

```diff
@@ -466,6 +466,13 @@ ptrdiff_t indexOf(Range)(Range s, in dchar c,
     return -1;
 }
 
+ptrdiff_t indexOf(T, size_t n)(ref T[n] s, in dchar c,
+        in CaseSensitive cs = CaseSensitive.yes) @safe pure
+    if (isSomeChar!T)
+{
+    return indexOf(s[], c, cs);
```

Comment:
An optimization would be to use `const(T)[]`, b/c of less template instantiations.